### PR TITLE
[PTFE-433] ✨ Add workflow cost metric

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

File: tests/api/metrics/test_job.py [hunk 1]
```diff
@@ -106,6 +106,24 @@ def test_job_cost(client, workflow_job, headers):
...
+    for line in metrics.text.split("\n"):
+        if "github_actions_workflow_cost_count_total{" in line:
+            assert "0.0" in line
+
+    workflow_job["workflow_job"]["conclusion"] = "success"
+    workflow_job["workflow_job"]["completed_at"] = "2021-11-29T14:59:57Z"
+    response = client.post("/webhook", json=workflow_job, headers=headers)
+    assert response.status_code == 202
+
+    response = client.post("/webhook", json=workflow_job, headers=headers)
```
[the remark this answers is not in the document]
Comment (gaspardmoindrot) [reply tests/api/metrics/test_job.py:121]: We don't actually run it twice; it's just that we simulate having a workflow with 2 jobs!

**Review by tcarmet — CHANGES_REQUESTED** [review]
I don't understand why this metric was added, as I fail to see the difference with `github_actions_job_cost_count` given that the label `workflow_name` is available on it as well.

Would you mind sharing the query you are trying to do?

**Comment by gaspardmoindrot** [comment]
> I don't understand why this metric was added, as I fail to see the difference with `github_actions_job_cost_count` given that the label `workflow_name` is available on it as well.
> 
> Would you mind sharing the query you are trying to do?

Certainly, in essence, what I wanted to do with the graph was to display the average cost per workflow, something like this:
```
sum(increase(github_actions_workflow_cost_count_total{repository=~"$repository", runner_type=~"$runner_type", repository_visibility=~"$repository_visibility", cloud=~"$cloud", workflow_name=~"$workflow_name"}[$__interval])) by (repository, workflow_name) / sum(increase(github_actions_workflow_total_count_total{repository=~"$repository", runner_type=~"$runner_type", repository_visibility=~"$repository_visibility", cloud=~"$cloud", workflow_name=~"$workflow_name"}[$__interval])) by (repository, workflow_name) > 0
```
As you know, github_actions_workflow_cost_count_total does not exist, so I wanted to add it. Initially, I tried to work around this issue by changing the query and finding an alternative approach, but without success. The information available to us is the cost of each job, the number of jobs, and the number of workflows. Based on that, I don't see how to calculate the average cost of each workflow. If you have any ideas, I'm open to them, and we can simply close this pull request.

**Comment by tcarmet** [comment]
Given how you are using this new metrics, you will reach the same result with the existing one:
```
sum(increase(github_actions_job_cost_count_total{repository=~"$repository", runner_type=~"$runner_type", repository_visibility=~"$repository_visibility", cloud=~"$cloud", workflow_name=~"$workflow_name"}[$__interval])) by (repository, workflow_name) / sum(increase(github_actions_workflow_total_count_total{repository=~"$repository", runner_type=~"$runner_type", repository_visibility=~"$repository_visibility", cloud=~"$cloud", workflow_name=~"$workflow_name"}[$__interval])) by (repository, workflow_name) > 0
```

Checkout how a graph can look like with it: https://mon.scality.net/grafana/d/WidbLgPnk/github-actions-monitoring?orgId=1&refresh=30s&viewPanel=72

**Pull request closed without merging**